internal/cache/file/cache_handler_test.go

```diff
@@ -62,457 +55,584 @@ type cacheHandlerTest struct {
 	cacheDir        string
 }
 
-func (chrT *cacheHandlerTest) SetupTest() {
+func initializeCacheHandlerTestArgs(t *testing.T, fileCacheConfig *config.FileCacheConfig, cacheDir string) *cacheHandlerTestArgs {
+	t.Helper()
 	locker.EnableInvariantsCheck()
-	chrT.cacheDir = path.Join(os.Getenv("HOME"), "cache/dir")
 
 	// Create bucket in fake storage.
-	chrT.fakeStorage = storage.NewFakeStorage()
-	storageHandle := chrT.fakeStorage.CreateStorageHandle()
-	chrT.bucket = storageHandle.BucketHandle(storage.TestBucketName, "")
+	fakeStorage := storage.NewFakeStorage()
+	storageHandle := fakeStorage.CreateStorageHandle()
+	bucket := storageHandle.BucketHandle(storage.TestBucketName, "")
 
 	// Create test object in the bucket.
 	testObjectContent := make([]byte, TestObjectSize)
 	_, err := rand.Read(testObjectContent)
-	require.NoError(chrT.T(), err)
-	chrT.object = chrT.getMinObject(TestObjectName, testObjectContent)
+	require.NoError(t, err)
+	object := createObject(t, bucket, TestObjectName, testObjectContent)
 
 	// fileInfoCache with testFileInfoEntry
-	chrT.cache = lru.NewCache(HandlerCacheMaxSize)
+	cache := lru.NewCache(HandlerCacheMaxSize)
 
 	// Job manager
-	chrT.jobManager = downloader.NewJobManager(chrT.cache, util.DefaultFilePerm,
-		util.DefaultDirPerm, chrT.cacheDir, DefaultSequentialReadSizeMb, &config.FileCacheConfig{
-			EnableCRC: true,
-		})
+	jobManager := downloader.NewJobManager(cache, util.DefaultFilePerm,
+		util.DefaultDirPerm, cacheDir, DefaultSequentialReadSizeMb, fileCacheConfig)
 
 	// Mocked cached handler object.
-	chrT.cacheHandler = NewCacheHandler(chrT.cache, chrT.jobManager, chrT.cacheDir, util.DefaultFilePerm, util.DefaultDirPerm)
+	cacheHandler := NewCacheHandler(cache, jobManager, cacheDir, util.DefaultFilePerm, util.DefaultDirPerm)
 
 	// Follow consistency, local-cache file, entry in fileInfo cache and job should exist initially.
-	chrT.fileInfoKeyName = chrT.addTestFileInfoEntryInCache(storage.TestBucketName, TestObjectName)
-	chrT.downloadPath = util.GetDownloadPath(chrT.cacheHandler.cacheDir, util.GetObjectPath(chrT.bucket.Name(), chrT.object.Name))
-	_, err = util.CreateFile(data.FileSpec{Path: chrT.downloadPath, FilePerm: util.DefaultFilePerm, DirPerm: util.DefaultDirPerm}, os.O_RDONLY)
-	require.NoError(chrT.T(), err)
-	_ = chrT.getDownloadJobForTestObject()
+	fileInfoKeyName := addTestFileInfoEntryInCache(t, cache, object, storage.TestBucketName)
+	downloadPath := util.GetDownloadPath(cacheHandler.cacheDir, util.GetObjectPath(bucket.Name(), object.Name))
+	_, err = util.CreateFile(data.FileSpec{Path: downloadPath, FilePerm: util.DefaultFilePerm, DirPerm: util.DefaultDirPerm}, os.O_RDONLY)
+	require.NoError(t, err)
+
+	job := jobManager.CreateJobIfNotExists(object, bucket)
+	require.NotNil(t, job)
+
+	t.Cleanup(func() {
+		fakeStorage.ShutDown()
```

Comment:
The cleanup should happen right after fakeStorage was created. Otherwise, if the test fails in between e.g. the precondition "require.NotNil" in the previous line is not satisfied, then fakeStorage.Shutdown won't be called. Same goes for the cacheDir too.

Reply:
Taken care in https://github.com/GoogleCloudPlatform/gcsfuse/pull/2178